src/antidote_pb_counter.erl

```diff
@@ -60,19 +60,33 @@ encode(Message) ->
 
 %% @doc process/2 callback. Handles an incoming request message.
 process(#fpbincrementreq{key=Key, amount=Amount}, State) ->
-    {ok, _Result} = antidote:append(Key, riak_dt_pncounter, {{increment, Amount}, node()}),
-    {reply, #fpboperationresp{success = true}, State};
+    case antidote:append(Key, riak_dt_pncounter, {{increment, Amount}, node()}) of
+        {ok, _Result} ->
+             {reply, #fpboperationresp{success = true}, State};
+        {error, Result} ->
+            lager:error("Operation failed due to ~p",[Result]),
+            {reply, #fpboperationresp{success = false}, State}
+    end;
 
 %% @doc process/2 callback. Handles an incoming request message.
 process(#fpbdecrementreq{key=Key, amount=Amount}, State) ->
-    {ok, _Result} = antidote:append(Key, riak_dt_pncounter, {{decrement, Amount}, node()}),
-    {reply, #fpboperationresp{success = true}, State};
+    case antidote:append(Key, riak_dt_pncounter, {{decrement, Amount}, node()}) of
+        {ok, _Result} ->
+            {reply, #fpboperationresp{success = true}, State};
+        {error, Result} ->
+            lager:error("Operation failed due to ~p",[Result]),
+            {reply, #fpboperationresp{success = false}, State}
+    end;    
 
-%% @doc process/2 callback. Handles an incoming request message.
 %% @todo accept different types of counters.
 process(#fpbgetcounterreq{key=Key}, State) ->
-    {ok, Result} = antidote:read(Key, riak_dt_pncounter),
-    {reply, #fpbgetcounterresp{value = Result}, State}.
+    case antidote:read(Key, riak_dt_pncounter) of
+        {ok, Result} ->
+            {reply, #fpbgetcounterresp{value = Result}, State};
+        {error, Reason} ->
+            lager:error("Operation failed due to ~p",[Reason]),
```

Comment:
I think this is still useful to figure out what is the error.. The current format of error message does not contain a field for the reason of failure, so we can only know if an operation succeeded or failed.. Though if needed, I can try to add more infor to message, which involves modifying files of riak_pb, antidote_pb and antidote...